core/shared/platform/linux-sgx/bh_platform.c

```diff
@@ -69,23 +70,48 @@ int bh_vprintf_sgx(const char * format, va_list arg)
     return 0;
 }
 
-void *
-bh_mmap(void *hint, unsigned int size, int prot, int flags)
+void* bh_mmap(void *hint, unsigned int size, int prot, int flags)
 {
-    /* TODO: implement bh_mmap in Linux SGX */
-    return NULL;
+    int mprot = 0;
+    unsigned alignedSize = (size+4095) & (unsigned)~4095; //Page aligned
+    void* ret = NULL;
+    sgx_status_t st = 0;
+
+    ret = sgx_alloc_rsrv_mem(alignedSize);
+    if (ret == NULL) {
+        bh_printf_sgx("bh_mmap(size=%d, alignedSize=%d, prot=0x%x) failed.",size, alignedSize, prot);
+        return NULL;
+    }
+    if (prot & MMAP_PROT_READ)
+        mprot |= SGX_PROT_READ;
+    if (prot & MMAP_PROT_WRITE)
+        mprot |= SGX_PROT_WRITE;
+    if (prot & MMAP_PROT_EXEC)
+        mprot |= SGX_PROT_EXEC;
+    st = sgx_tprotect_rsrv_mem(ret, alignedSize, mprot);
+    if (st != SGX_SUCCESS) bh_printf_sgx("bh_mmap(size=%d,prot=0x%x) failed to set protect.",size, prot);
```

Comment:
Should check return value, if failed, free memory and return NULL.

Reply:
OK, it is reasonable comment.